operator/webhook/core/sync.py

```diff
@@ -453,6 +454,79 @@ def _new_actuator_service(parent):
     return service
 
 
+def _compute_status(parent: Mapping, children: Mapping) -> Mapping:
+    route_name = parent["metadata"]["name"]
+    expected_replicas = parent["spec"]["replicas"]
+
+    init_status = {
+        "expectedReplicas": expected_replicas,
+        "readyReplicas": 0,
+        "runningReplicas": 0,
+    }
+
+    deployments = children["Deployment.apps/v1"]
+
+    if route_name not in deployments:
+        return init_status
+
+    deployment_status = deployments[route_name].get("status")
+
+    if not deployment_status:
+        return init_status
+
+    ready_replicas = deployment_status.get("readyReplicas", 0)
+
+    available_conditions = [
+        c for c in deployment_status["conditions"] if c["type"] == "Available"
+    ]
+
+    ready_conditions = [
+        _get_status_ready_condition(
+            parent["status"], expected_replicas == ready_replicas
+        )
+    ]
+
+    return {
+        "expectedReplicas": expected_replicas,
+        "readyReplicas": ready_replicas,
+        "runningReplicas": deployment_status.get("replicas", 0),
+        "conditions": available_conditions + ready_conditions,
+    }
+
+
+def _get_status_ready_condition(parent_status: Mapping, is_ready: bool) -> Mapping:
+    condition_type = "Ready"
+
+    ready_condition_list = (
+        [c for c in parent_status["conditions"] if c["type"] == condition_type]
+        if parent_status
+        else {}
+    )
+
+    ready_condition = ready_condition_list[0] if ready_condition_list else None
+    if ready_condition and ready_condition.get("status") == str(is_ready):
+        return ready_condition
+
+    transition_time = datetime.now(timezone.utc).strftime("%Y-%m-%dT%H:%M:%SZ")
+
+    if is_ready:
```

Comment:
📓 nitpick but we could create a dictionary then add a field and return it. Might help reduce a little redundancy and keep it consistent later. Tradeoff is that its mutated for a period of time.

```
condition = {
            "lastTransitionTime": transition_time,
            "reason": "ReplicasNotReady",
            "status": str(is_ready),
            "type": condition_type,
        }
if is_ready:
    condition['message'] = "All IntegrationRoute pod replicas are ready"
    condition['reason'] = "ReplicasReady"
else:
    condition['message'] = "Some IntegrationRoute pod replicas are not ready"
    condition['reason'] = "ReplicasNotReady"
return condition
```

Reply:
Updated